akka-http/src/main/scala/akka/http/scaladsl/server/directives/FileUploadDirectives.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.http.scaladsl.server.directives
+
+import java.io.File
+
+import akka.stream.scaladsl._
+import akka.http.scaladsl.server.Directive1
+import akka.http.scaladsl.model.{ ContentType, Multipart }
+import akka.util.ByteString
+import scala.util.Try
+
+trait FileUploadDirectives {
+
+  import BasicDirectives._
+  import FutureDirectives._
+  import MarshallingDirectives._
+
+  /**
+   * Collects the bytes of the first file part from the request into the given destination file
+   *
+   * @param destination Will be overwritten if not empty
+   */
+  def uploadedFile(destination: File): Directive1[Try[UploadedFile]] =
```

Comment:
We could expect the destination to be a folder and reuse the `part.fileName` as the destination file. Another option would be:
```scala
def upload(fieldNames: String*, destination: FileInfo => File): Directive1[Map[FileInfo, Try[File]]]
```